packages/agreement-process/test/testAgreementConsumerDocuments.ts

```diff
@@ -1,97 +1,265 @@
+      });
+
+      it("should throw an agreementDocumentNotFound error when the document does not exist", async () => {
+        const authData = getRandomAuthData(agreement1.consumerId);
+        const agreementDocumentId = generateId<AgreementDocumentId>();
+        await expect(
+          agreementService.getAgreementConsumerDocument(
+            agreement1.id,
+            agreementDocumentId,
+            authData
+          )
+        ).rejects.toThrowError(
+          agreementDocumentNotFound(agreementDocumentId, agreement1.id)
+        );
+      });
     });
 
-    it("should throw an agreementDocumentNotFound error when the document does not exist", async () => {
-      const authData = getRandomAuthData(agreement1.consumerId);
-      const agreementDocumentId = generateId<AgreementDocumentId>();
-      await expect(
-        agreementService.getAgreementConsumerDocument(
-          agreement1.id,
-          agreementDocumentId,
-          authData
-        )
-      ).rejects.toThrowError(
-        agreementDocumentNotFound(agreementDocumentId, agreement1.id)
-      );
+    describe("add", () => {
+      it("should succeed on happy path", async () => {
+        const organizationId = generateId<TenantId>();
+        const authData: AuthData = {
+          ...getRandomAuthData(organizationId),
+          userRoles: ["admin"],
+        };
+        const agreement = getMockAgreement(
+          generateId<EServiceId>(),
+          organizationId
+        );
+
+        const consumerDocument = getMockConsumerDocument(agreement.id);
+        const correlationId = "9999";
+
+        await addOneAgreement(agreement, postgresDB, agreements);
+
+        await agreementService.addConsumerDocument(
+          agreement.id,
+          consumerDocument,
+          authData,
+          correlationId
+        );
+        const { data: payload } = await readLastAgreementEvent(
+          agreement.id,
+          postgresDB
+        );
+
+        const actualConsumerDocument = decodeProtobufPayload({
+          messageType: AgreementConsumerDocumentAddedV1,
+          payload,
+        });
+
+        expect(actualConsumerDocument).toMatchObject({
+          agreementId: agreement.id,
+          document: {
+            ...consumerDocument,
+            createdAt: actualConsumerDocument.document?.createdAt,
+          },
+        });
+      });
+
+      it("should throw an agreementNotFound error when the agreement does not exist", async () => {
+        const organizationId = generateId<TenantId>();
+        const authData: AuthData = {
+          ...getRandomAuthData(organizationId),
+          userRoles: ["admin"],
+        };
+        const agreement = getMockAgreement(
+          generateId<EServiceId>(),
+          organizationId
+        );
+
+        const consumerDocument = getMockConsumerDocument(agreement.id);
+        const correlationId = "9999";
+
+        await addOneAgreement(agreement, postgresDB, agreements);
```

Comment:
Since you just need to add a random other agreement without any constraint, you could just replace all these lines with the following: 

``` 
      await addOneAgreement(getMockAgreement(), postgresDB, agreements);
      const authData = getRandomAuthData();
```